watchtower/wtclient/client.go

```diff
@@ -777,6 +865,277 @@ func (c *TowerClient) nextSessionQueue() (*sessionQueue, error) {
 	return c.getOrInitActiveQueue(candidateSession, updates), nil
 }
 
+// handleChannelCloses listens for channel close events and marks channels as
+// closed in the DB.
+//
+// NOTE: This method MUST be run as a goroutine.
+func (c *TowerClient) handleChannelCloses(chanSub subscribe.Subscription) {
+	defer c.wg.Done()
+
+	c.log.Tracef("Starting channel close handler")
+	defer c.log.Tracef("Stopping channel close handler")
+
+	for {
+		select {
+		case event, ok := <-chanSub.Updates():
+			if !ok {
+				log.Debugf("channel notifier has exited")
+				return
+			}
+
+			switch event := event.(type) {
+			case channelnotifier.ClosedChannelEvent:
+				closeInfo := event.CloseSummary
+				chanID := lnwire.NewChanIDFromOutPoint(
+					&closeInfo.ChanPoint,
+				)
+
+				// We only care about channels registered with
+				// the tower client.
+				c.backupMu.Lock()
+				_, ok := c.summaries[chanID]
+				c.backupMu.Unlock()
+				if !ok {
+					continue
+				}
+
+				sl, err := c.cfg.DB.MarkChannelClosed(
+					chanID, closeInfo.CloseHeight,
+				)
+				if err != nil {
+					log.Errorf("could not handle closed "+
+						"channel(%s): %v", chanID, err)
+					continue
+				}
+
+				for _, sID := range sl {
+					delay, err := newRandomDelay(
```

Comment:
"for given session ID and block height, compute random delay and push into closableSessionQueue" is duplicated (line 618)